### PR TITLE
Content Negotiation for CoAP JS Thing

### PR DESCRIPTION
**Pull request by SergioCasCeb**
- Modified the CoAP Thing to include headers for content negotiation (Text, JSON, and CBOR)
- Reflected such information on the generated TD
- Created a simple client to test the functionality of the Thing

### CODE REVIEW

File: things/calculator/coap/client/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# CoAP Test Client
+
+This client was created with the sole purpose of testing the main functionality of the CoAP Calculator Thing specifically the content negotiation.
+
+**This folder and files are not final and might be subject to changes or might be completely removed in the future.**
+
+## Enpoints
```
Comment (danielpeintner) [inline things/calculator/coap/client/README.md:7]:
```suggestion
## Endpoints
```

File: things/calculator/coap/client-tests/content-negotiation-coap-client.js [hunk 2]
```diff
@@ -18,28 +18,27 @@ const fullTDEndpoint = `/${thingName}`,
 /****************************************/
 
 // GET request to retrieve thing description
-function getFullTD() {
+function getFullTD(acceptType) {
```
Comment (danielpeintner) [inline things/calculator/coap/client-tests/content-negotiation-coap-client.js:21]: Maybe my personal opinion but I think that the functions should expect the _full_ accept value like
`application/json`

File: things/calculator/coap/client-tests/content-negotiation-coap-client.js [hunk 3]
```diff
@@ -267,23 +249,21 @@ function addNumber() {
 /****************************************/
 
 // POST request to perform the subtract action
-function subtractNumber() {
+function subtractNumber(acceptType, contentType, numberToSubtract) {
```
Comment (danielpeintner) [inline things/calculator/coap/client-tests/content-negotiation-coap-client.js:252]: Similar here... with contentType

File: things/calculator/coap/client-tests/content-negotiation-coap-client.js (hunk 2 above)
Comment (SergioCasCeb) [reply things/calculator/coap/client-tests/content-negotiation-coap-client.js:21]: I was debating the same while doing it. Thanks for the feedback, I will change it.

**Review by egekorkan — APPROVED** [review]

**Pull request merged**